ush/sub_gaea

```diff
@@ -0,0 +1,166 @@
+    t) timew=$OPTARG;;
+    u) userid=$OPTARG;;
+    v) verbose=YES;;
+    w) when=$OPTARG;;
+    \?) echo $0: invalid option >&2;echo "$usage" >&2;exit 1;;
+  esac
+done
+shift $(($OPTIND-1))
+if [[ $# -eq 0 ]];then
+  echo $0: missing executable name >&2;echo "$usage" >&2;exit 1
+fi
+exec=$1
+if [[ ! -s $exec ]]&&which $exec >/dev/null 2>&1;then
+  exec=$(which $exec)
+fi
+shift
+args="$*"
+bn=$(basename $exec)
+export jobname=${jobname:-$bn}
+output=${output:-$jobname.out}
+myuser=$LOGNAME
+myhost=$(hostname)
+
+if [ -d /lustre/f2/scratch/$LOGNAME ]; then
+  DATA=/lustre/f2/scratch/$LOGNAME/tmp
+fi
+DATA=${DATA:-$ptmp/tmp}
+
+mkdir -p $DATA
+
+queue=${queue:-batch}
+timew=${timew:-01:20:00}
+task_node=${task_node:-$procs}
+export size=$((nodes*task_node))
+echo "In sub_gaea: task_node, nodes, size=",$task_node,$nodes,$size
+envars=$envars
+threads=${rcpu:-1}
+
+export TZ=GMT
+cfile=$DATA/sub$$
+> $cfile
+echo "#!/bin/bash -l"                                                           >> $cfile
+echo ""                                                                            >> $cfile
+echo "#SBATCH --output=$output"                                                    >> $cfile
+echo "#SBATCH --job-name=$jobname"                                                 >> $cfile
+echo "#SBATCH --qos=normal"                                                        >> $cfile
+echo "#SBATCH --clusters=c4"                                                       >> $cfile
+echo "#SBATCH --time=$timew"                                                       >> $cfile
+echo "#SBATCH --nodes=$nodes --ntasks-per-node=$procs --cpus-per-task=$threads"    >> $cfile
+echo "#SBATCH --account=nggps_emc"                                                    >> $cfile
+echo "#SBATCH --mem=0"                                                             >> $cfile
+
+echo "" >>$cfile
+echo "export OMP_NUM_THREADS=$threads"      >> $cfile
+echo "" >>$cfile
+echo ". "$(awk '{ print $1, $2, $3, $4, $5, $6, $7, $8, $9 }' $regdir/regression_var.out)  >>$cfile
+echo "" >>$cfile
+
+echo "source /lustre/f2/dev/role.epic/contrib/Lmod_init.sh"  >> $cfile
+echo "module use $gsisrc/modulefiles"                        >> $cfile
+echo "module load gsi_gaea" >> $cfile
+echo "module list"                                           >> $cfile
+echo "" >>$cfile
+
+cat $exec >> $cfile
+
+if [[ $nosub = YES ]];then
+  cat $cfile
+  exit
+elif [[ $verbose = YES ]];then
+  set -x
+  cat $cfile
+fi
+
+if [[ $stdin = YES ]];then
+  cat
+fi >>$cfile
+if [[ $nosub = YES ]];then
+  cat $cfile
+  exit
+elif [[ $verbose = YES ]];then
+  set -x
+  cat $cfile
+fi
+sbatch=${sbatch:-sbatch}
+
+ofile=$DATA/subout$$
+>$ofile
+chmod 777 $ofile
+$sbatch $cfile >$ofile
+rc=$?
+cat $ofile
+if [[ -w $SUBLOG ]];then
+  jobn=$(grep -i submitted $ofile|head -n1|cut -d\" -f2)
+  date -u +"%Y%m%d%H%M%S : $subcmd : $jobn" >>$SUBLOG
+fi
+#rm $cfile $ofile
```

Comment:
Uncomment these two lines used for debugging.

Reply:
uncommented in the updated version